### PR TITLE
Fix/tests fixes

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

some tests fixes

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/test_manager/test_manager.py [hunk 1]
```diff
@@ -357,6 +357,12 @@ def test_start_all(self, *args):
             period=0.5,
         )
 
+        wait_for_condition(
+            lambda: "Start processing messages"
+            in open(self.log_file(self.agent_name), "r").read(),
+            timeout=60,
```
Comment (5A11) [inline tests/test_manager/test_manager.py:363]: Introduce a constant `DEFAULT_TIMEOUT = 60` and use across.

File: tests/test_manager/test_manager.py [hunk 2]
```diff
@@ -457,6 +463,11 @@ def test_remove_running_agent(self, *args):
         """Test fail on remove running agent."""
         self.test_start_all()
 
+        wait_for_condition(
+            lambda: "Start processing messages"
+            in open(self.log_file(self.agent_name), "r").read(),
+            timeout=60,
```
Comment (5A11) [inline tests/test_manager/test_manager.py:469]: same as above

File: tests/test_manager/test_utils.py [hunk 3]
```diff
@@ -37,26 +37,26 @@ def _process_exception():
...
+def _process_return_value(value_to_return):
+    return value_to_return
 
 
 def test_process_run_in_venv_timeout_error():
     """Test timeout error raised for process running too long."""
     with TemporaryDirectory() as tmp_dir:
         with pytest.raises(TimeoutError):
-            run_in_venv(tmp_dir, _process_long, timeout=3)
+            run_in_venv(tmp_dir, _process_long, timeout=20)
```
Comment (5A11) [inline tests/test_manager/test_utils.py:48]: Introduce a constant DEFAULT_TIMEOUT = 20 and use across.

File: tests/test_manager/test_utils.py [hunk 4]
```diff
@@ -37,26 +37,26 @@ def _process_exception():
...
 
 
 def test_process_run_in_venv_raise_custom_exception():
     """Test process returns expcetion."""
     with TemporaryDirectory() as tmp_dir:
         with pytest.raises(Exception, match="Expected"):
-            run_in_venv(tmp_dir, _process_exception, timeout=3)
+            run_in_venv(tmp_dir, _process_exception, timeout=20)
```
Comment (5A11) [inline tests/test_manager/test_utils.py:55]: Same as above

File: tests/test_manager/test_utils.py [hunk 5]
```diff
@@ -37,26 +37,26 @@ def _process_exception():
...
 
 
 def test_process_run_in_venv_return_value():
     """Test process return value."""
     with TemporaryDirectory() as tmp_dir:
-        ret_value = run_in_venv(tmp_dir, _process_return_value, timeout=3)
+        ret_value = run_in_venv(tmp_dir, _process_return_value, 20, RETURN_VALUE)
```
Comment (5A11) [inline tests/test_manager/test_utils.py:61]: Same as above

File: aea/connections/base.py [hunk 6]
```diff
@@ -277,12 +277,12 @@ def from_config(
         """
         configuration = cast(ConnectionConfig, configuration)
         directory = cast(Path, configuration.directory)
-        load_aea_package(configuration)
         connection_module_path = directory / "connection.py"
         if not (connection_module_path.exists() and connection_module_path.is_file()):
             raise AEAComponentLoadException(
                 "Connection module '{}' not found.".format(connection_module_path)
             )
+        load_aea_package(configuration)
```
Comment (DavidMinarsch) [inline aea/connections/base.py:285]: Can you explain?

File: aea/manager/manager.py [hunk 7]
```diff
@@ -261,13 +267,16 @@ def _run_agent(
...
+        asyncio.set_event_loop(asyncio.new_event_loop())
         try:
             aea = agent_alias.get_aea_instance()
 
             def stop_event_thread() -> None:
                 try:
                     while run_stop_thread:
-                        if stop_event.wait(0.05) is True:
+                        if stop_event.wait(0.01) is True:
+                            print("Stop event set!")
```
Comment (DavidMinarsch) [inline aea/manager/manager.py:279]: why not logging module?

**Review by DavidMinarsch — COMMENTED** [review]
Tests on this PR are starting to look very promising! Only issue: Windows tests all time out..

File: aea/manager/manager.py (hunk 7 above)
Comment (solarw) [reply aea/manager/manager.py:279]: it's a subprocess, not sure about logging. anyway it was added for debug purposes and i can remove it now

File: aea/connections/base.py (hunk 6 above)
Comment (solarw) [reply aea/connections/base.py:285]: it's done for one test. with load_aea_package update, i perform extra check for dir exists and one negative test fails

now it's ok cause connection module check done before the module loading attempt.

File: tests/test_manager/test_manager.py (hunk 2 above)
Comment (solarw) [reply tests/test_manager/test_manager.py:469]: done

File: tests/test_manager/test_manager.py (hunk 1 above)
Comment (solarw) [reply tests/test_manager/test_manager.py:363]: done

File: tests/test_manager/test_utils.py (hunk 3 above)
Comment (solarw) [reply tests/test_manager/test_utils.py:48]: done

File: tests/test_manager/test_utils.py (hunk 4 above)
Comment (solarw) [reply tests/test_manager/test_utils.py:55]: done

File: tests/test_manager/test_utils.py (hunk 5 above)
Comment (solarw) [reply tests/test_manager/test_utils.py:61]: done

**Comment by solarw** [comment]
> Tests on this PR are starting to look very promising! Only issue: Windows tests all time out..

not sure can fix it, looks like it's just tests execution time, tests canceled on different cases, not stuck on something one

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Review by 5A11 — APPROVED** [review]
lgtm

**Pull request merged**